### ISSUE TITLE
"list[tuple[Literal[1]]]" is incompatible with "list[tuple[int]]" since 1.1.315

### ISSUE THREAD

**Issue by moskupols** [microsoft/pyright#5485]
**Describe the bug**
Pyright doesn't treat Literal[{integer value}] compatible with int sometimes.

**Code or Screenshots**
```python
def take_tuple(t: list[tuple[int]]):
    pass

the_tuple = [(1,)]
take_tuple(the_tuple)
```

**To Reproduce**
Run pyright in basic mode on the code snippet above. Here's the error I'm getting:
```
wut.py:5:12 - error: Argument of type "list[tuple[Literal[1]]]" cannot be assigned to parameter "t" of type "list[tuple[int]]" in function "take_tuple"
    "list[tuple[Literal[1]]]" is incompatible with "list[tuple[int]]"
      TypeVar "_T@list" is invariant
        "tuple[Literal[1]]" is incompatible with "tuple[int]"
          Tuple entry 1 is incorrect type
            "Literal[1]" is incompatible with "int"
```
**Expected behavior**
There's no error reported

**VS Code extension or command-line**
pyright command-line via python package, I was able to reproduce this in versions 1.1.315 through 1.1.317 but not 1.1.314
python version is 3.11

**Additional context**
Note that the following snippets pass checks:
```python
def take_tuple(t: list[tuple[int]]):
    pass

take_tuple([(1,)])
```
```python
def take_tuple(t: tuple[int]):
    pass

t = (1,)
take_tuple((1,))
```

**Comment by erictraut** [comment]
The current behavior is correct. The type parameter for `list` is invariant, so the type arguments must match exactly. The type `list[Literal[1]]` is not a subtype of `list[int]`.  

There was previously a bug where pyright didn't report this incompatibility. This bug was fixed recently, which is why you see the change in behavior.
